ansible/roles/providers/tasks/misc.yml

```diff
@@ -57,7 +57,7 @@
     register: system_cloud_init_cmd
   - name: Set cloud-init version fact
     set_fact:
-      system_cloud_init_version: "{{ system_cloud_init_cmd.stdout | regex_replace('^.*cloud-init\\s(?P<version>\\d+?\\.?\\d+)-.*$', '\\g<version>') }}"
+      system_cloud_init_version: "{{ system_cloud_init_cmd.stdout | regex_replace('^.*cloud-init\\s(?P<version>\\d+?\\.?\\d+)(-.*)?$', '\\g<version>') }}"
```

Comment:
regex tested here https://regex101.com/r/2Xc5Vx/1

Reply:
thank you for posting the regex test, awesome.